pkg/nodeconfig/nodeconfig.go

```diff
@@ -169,6 +169,17 @@ func (nc *nodeConfig) Configure() error {
 		if err := drain.RunCordonOrUncordon(drainHelper, nc.node, true); err != nil {
 			nc.log.Info("unable to cordon", "node", nc.node.GetName(), "error", err)
 		}
+
+		// Ensure we are annotating the node as soon as the Node object is created, so that we can identify which
+		// controller should be watching it
+		nc.addAdditionalAnnotations()
+		nc.addPubKeyHashAnnotation()
+		node, err := nc.k8sclientset.CoreV1().Nodes().Update(context.TODO(), nc.node, meta.UpdateOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "error updating annotations on node %s", nc.node.GetName())
```

Comment:
Given we are updating different annotations in different places please make the message `error updating additional / public key hash annotations...` or something to that effect.